deleter/ec2.go

```diff
@@ -1154,12 +1168,106 @@ func (rd *EC2NatGatewayDeleter) DeleteResources(cfg *DeleteConfig) error {
 			return err
 		}
 
+	}
+
+	if cfg.DryRun {
+		return nil
+	}
+
+	// If we don't wait until nat gateways are deleted, EIP disassociation/release
+	// and customer gateway disassociation/deletion will fail
+	fmt.Println("Waiting for EC2 NAT Gateways to delete...")
+	deletedNGWs, aliveNGWs, err := rd.waitUntilNatGatewaysDeleted(ngws)
```

Comment:
Is it the case that _always_ delete will fail if the nat gateway aren't deleted?

I'm wondering if it makes sense to incorporate this as part of retrying instead of blocking